scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/testmode/TestInterpreterRunner.scala

```diff
@@ -12,7 +12,9 @@ object TestInterpreterRunner {
    * and in this case there is loaders conflict
    */
   private[engine] def testResultsVariableEncoder: Any => io.circe.Json = {
-    lazy val encoder = ToJsonEncoder(failOnUnknown = false, Thread.currentThread().getContextClassLoader)
+    // Variables keep some intermediate state of a scenario. We can't be sure that there are only types that are supported
+    // for json encoding. It is better to print not-well formatted .toString result than fail in this case
+    lazy val variablesEncoder = new ToJsonEncoder(failOnUnknown = false, Thread.currentThread().getContextClassLoader)
```

Comment:
Perhaps there can be a method ToJsonEncoder.looseEncoder(classLoader) for cases like that, where custom loader is used?